libraries/ms-common/inc/adc.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+// Generic ADC Driver
+
+#include <stdbool.h>
+#include "gpio.h"
+#include "status.h"
+
+
+typedef enum {
+  ADC_MODE_SINGLE = 0,
+  ADC_MODE_CONTINUOUS,
+} ADCMode;
+
+typedef enum {
+  ADC_SAMPLE_RATE_1 = 0,
```

Comment:
I still don't understand what this is for without reading through the datasheet.